.github/actions/pre_test_setup/action.yml

```diff
@@ -0,0 +1,49 @@
+name: pre test Setup
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+
+    - name: Use Node.js ${{ inputs.node-version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node-version }}
+
+    - name: Enable corepack
+      shell: bash
```

Comment:
Is this necessary now? It wasn't there before. It makes sense to specify the shell when you need something that the shell provides, but not to just run a binary, as in this context.